source/compiler/sc5.c

```diff
@@ -422,3 +424,67 @@ int pc_geterrorwarnings()
   return errwarn;
 }
 
+/* static_check()
+ * checks compile-time assertions and triggers an error or warning
+ *
+ * Parameters:
+ * FALSE for error if assertion fails
+ * TRUE for warning if assertion fails
+ */
+int static_check(int warning)
```

Comment:
Is sc5.c a good place for this function? I don't see any other parsing business in this file, it looks like it is intended to hold the error reporting system, parsing should be done elsewhere. 

sc1.c might be a better place, we already have `doassert` in there. Also, how about renaming this function to `do_static_assert`? So that it's more consistent with other functions.